homeassistant/components/melcloud/water_heater.py

```diff
@@ -0,0 +1,133 @@
+"""Platform for water_heater integration."""
+from typing import List, Optional
+
+from pymelcloud import DEVICE_TYPE_ATW, AtwDevice
+from pymelcloud.atw_device import (
+    PROPERTY_OPERATION_MODE,
+    PROPERTY_TARGET_TANK_TEMPERATURE,
+)
+from pymelcloud.device import PROPERTY_POWER
+
+from homeassistant.components.water_heater import (
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    WaterHeaterDevice,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import DOMAIN, MelCloudDevice
+from .const import ATTR_STATUS, TEMP_UNIT_LOOKUP
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+):
+    """Set up MelCloud device climate based on config_entry."""
+    mel_devices = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            AtwWaterHeater(mel_device, mel_device.device)
+            for mel_device in mel_devices[DEVICE_TYPE_ATW]
+        ],
+        True,
+    )
+
+
+class AtwWaterHeater(WaterHeaterDevice):
+    """Air-to-Water water heater."""
+
+    def __init__(self, api: MelCloudDevice, device: AtwDevice):
+        """Initialize water heater device."""
+        self._api = api
+        self._device = device
+        self._name = device.name
+
+    async def async_update(self):
+        """Update state from MELCloud."""
+        await self._api.async_update()
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return a unique ID."""
+        return f"{self._api.device.serial}-{self._api.device.mac}"
```

Comment:
Same question about mac.

Reply:
Here I can simply drop it without issues.